lib/packages.js

```diff
@@ -467,3 +469,34 @@ function _isDir(filename) {
   } catch(e) {}
   return stat ? stat.isDirectory() : false;
 }
+
+function _detectCycle(cache) {
+  const initialPkgName = Object.keys(cache)[0];
+  const visited = new Set();
+  const recStack = {};
+
+  _detectCylceUtil(cache, initialPkgName, visited, recStack);
+}
+
+function _detectCylceUtil(cache, name, visited, recStack) {
+  if(!visited.has(name)) {
+    visited.add(name);
+    recStack[name] = true;
+
+    const deps = Object.keys(cache[name].manifest.dependencies || {});
+    for(const dep of deps) {
+      if(!(dep in cache)) {
+        continue;
+      }
+      if(!visited.has(dep)) {
+        _detectCylceUtil(cache, dep, visited, recStack);
+      } else if(recStack[dep]) {
+        const msg = `Cycle in Depedencies: ${Object.keys(recStack)
+          .filter(dep => recStack[dep] === true)}`;
+        throw new Error(msg);
```

Comment:
Might be easier to read and edit if code moved out of the string (making up a var name here):
```js
const cyc = Object.keys(recStack).filter(dep => recStack[dep] === true);
throw new Error(`Cycle in Depedencies: ${cyc}`;
```